server/src/main/java/org/folio/shared/index/storage/Storage.java

```diff
@@ -179,6 +179,20 @@ Future<Void> deleteBibRecord(SqlConnection conn, String localIdentifier, UUID so
         .mapEmpty());
   }
 
+  Future<Void> upsertGlobalRecord(UUID sourceId, JsonObject globalRecord, JsonArray matchKeyConfigs,
+      int iter) {
+
+    return pool.withTransaction(conn ->
+            upsertGlobalRecord(conn, sourceId, globalRecord, matchKeyConfigs)
+        )
+        .recover(x -> {
+          if (iter > 2) {
+            return Future.failedFuture(x);
+          }
+          return upsertGlobalRecord(sourceId, globalRecord, matchKeyConfigs, iter + 1);
```

Comment:
@adamdickmeiss Why do we have two iterations?

Reply:
Because of collision with new cluster value. It's a rare occurrence

Reply:
See what was removed in this PR

Reply:
Is this the Optimistic Locking approach?